cumulus_etl/upload_notes/cli.py

```diff
@@ -341,16 +341,16 @@ async def upload_notes_main(args: argparse.Namespace) -> None:
     store.set_user_fs_options(vars(args))
 
     root_input = store.Root(args.dir_input)
-    codebook = deid.Codebook(args.dir_phi)
 
     # Auth & read files early for quick error feedback
     client = fhir.create_fhir_client_for_cli(args, root_input, ["DocumentReference"])
     access_token = common.read_text(args.ls_token).strip()
     labels = ctakesclient.filesystem.map_cui_pref(args.symptoms_bsv)
 
     async with client:
-        ndjson_folder = await gather_docrefs(client, root_input, codebook, args)
-        notes = await read_notes_from_ndjson(client, ndjson_folder.name, codebook)
+        with deid.Codebook(args.dir_phi) as codebook:
+            ndjson_folder = await gather_docrefs(client, root_input, codebook, args)
+            notes = await read_notes_from_ndjson(client, ndjson_folder.name, codebook)
```

Comment:
This here is the bug fix - we hadn't been saving the codebook. In practice, this is unlikely to surface a bug, since you are likely to be working with documents you already ETL'd... but :shrug: let's not leave this gap here.

Reply:
Hmm OK fair. I guess in my head the user doesn't need to care that it saves. It's like flushing an opened file at the end of the context manager - just "cleanup" that happens behind the scene.

But certainly I can add a comment/docstring to the codebook class about usage and what happens.

Reply:
Added this to the class docstring:
```
    Example usage as a context manager:

    with Codebook('/path/to/phi/') as codebook:
        print(codebook.fake_id("Patient", "1234"))

    At the end of the context manager, `save()` is called, which will cache mappings of any new
    patient/encounter IDs and preserve the hashing salt used to generate anonymized IDs (if given
    a fresh PHI folder).

    You can also use this class outside of a context manager, just be careful to call `save()`
    yourself.
```